src/cart/crt_rpc.c

```diff
@@ -928,7 +928,7 @@ crt_client_get_contact_rank(crt_context_t crt_ctx, crt_group_t *grp,
 		*ret_idx = -1;
 	} else {
 		/* Pick random rank from the list */
-		*ret_idx = rand() % membs->rl_nr;
+		*ret_idx = D_RAND() % membs->rl_nr;
```

Comment:
[Question] Given the number of "%" usages throughout our code, should we consider some non-linear-congruential algorithm (e.g., random_r(3)), or introducing a

    /** [0, n) */
    long int d_rand_n(long int n);

that uses "/" instead of "%"? (Probably out of the scope of this ticket.)

Reply:
@liw yeah, probably best for another ticket. I am not that familiar with this issue in particular though so I can't say for sure.